### ISSUE TITLE
latexmk fails to detect dependencies when -outdir path has a trailing slash

### ISSUE THREAD

**Issue by AlessandroVol23** [James-Yu/LaTeX-Workshop#1114]
Hi,

I've got a problem with the extension Latex Workshop in VS Code. I've updated to VS Code Version 1.30.2 (1.30.2) and LaTeX workshop version 5.20.1 and after that the problem started immediately.

My setup looks like following:

- root/
- main.tex
- chapter/
    - chapter_1.tex

The extension just compiles files when the main.tex file changes. Files in the subfolder chapter which I included with the ``input`` command don't compile when they change anymore. For example I change something in chapter_1.tex and hit save the output says: 

**Logs**

    [16:05:42] Found root file in root directory: main.tex
    [16:05:42] Root file remains unchanged from: main.tex.
    [16:05:42] Building root file: main.tex
    [16:05:42] Build root file main.tex
    [16:05:42] Recipe step 1: latexmk, -synctex=1,-interaction=nonstopmode,-file-line-error,-pdf,-outdir=./,/
    [16:05:42] File watcher: responding to change in chapter_1.tex
    [16:05:42] chapter_1.tex
    [16:05:42] Recipe of length 1 finished.
    [16:05:42] Successfully built main.tex

(I've deleted the folder paths)

It isn't reproducible if I create another folder with the same structure and edit something in the file in the sub chapter it is working. 

I've tried to include the magic command ``% !TeX root = ../main.tex`` -> Didn't help

And I've tried to reinstall the extension which didn't help either. 

Thanks for your help! 

PS: Not sure where to post it (stackoverflow, tex.stackexchange or as an issue here). Because it happened after an update I post it as an issue here. 

**Expected behavior**
Save file in sub-directory -> Latex file gets compiled -> Preview of new file.

**Desktop (please complete the following information):**
 - OS: Mac OS High Sierra 10.13.6
 - VS Code version: 1.30.2 (1.30.2)
 - Extension version: 5.20.1

**Comment by jlelong** [comment]
The logs confirm that the build command is indeed called when `chapter_1.tex` is saved.
The problems comes from `latexmk`. If you modify `chapter_1.tex` and call `latexmk` manually with 
```
latexmk -synctex=1 -interaction=nonstopmode -file-line-error -pdf -outdir=. main
```
it simply responds that all targets are up-to-date. When removing the `-outdir` option, everything works as expected. You need to clean all auxiliary files when removing the `-outdir` option.

Surprisingly, the dependencies listed in the `main.fdb_latexmk` file depends on the use of `-outdir` option or not.

**Comment by AlessandroVol23** [comment]
> The logs confirm that the build command is indeed called when `chapter_1.tex` is saved.
> The problems comes from `latexmk`. If you modify `chapter_1.tex` and call `latexmk` manually with
> 
> ```
> latexmk -synctex=1 -interaction=nonstopmode -file-line-error -pdf -outdir=./  main
> ```
> it simply responds that all targets are up-to-date. The culprit is the trailing slash in the path given
> to `-outdir`. If you run
> 
> ```
> latexmk -synctex=1 -interaction=nonstopmode -file-line-error -pdf -outdir=.  main
> ```
> everything works as expected.
> You need to clean all auxiliary files when changing the `-outdir` option.

Thanks! I am not sure how to edit this in the settings. Currently it is in the settings like that:

```
  "latex-workshop.latex.tools": [
    {
      "name": "latexmk",
      "command": "latexmk",
      "args": [
        "-synctex=1",
        "-interaction=nonstopmode",
        "-file-line-error",
        "-pdf",
        "-outdir=%OUTDIR%",
        "%DOC%"
      ]
    },
```

Should I just override `"-outdir=%OUTDIR%",` to `"-outdir=.` in the user settings or what is the best way to change the parameter?

**Comment by jlelong** [comment]
It will be fixed in the next release. The variable to change is `latex-workshop.latex.outputDir`.

**Comment by nasosev** [comment]
This issue is still not fixed for me (as of 5.21.0). Nothing after 5.19.0 works.
